deep_water_level/src/model.py

```diff
@@ -9,31 +10,41 @@ def __init__(self, image_size: Tuple[int, int, int] = (3, 810, 510)):
         super().__init__()
 
         self.image_size = image_size
-        # Calculate the target parameter size after the convolutions and poolings
-        self.linear_line_size = (image_size[1] // 4 // 4) * (image_size[2] // 4 // 4) * 12
 
         self.conv1 = nn.Conv2d(in_channels=image_size[0], out_channels=6, kernel_size=4, stride=2, padding=1)
-        self.pool1 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.pool1 = nn.MaxPool2d(kernel_size=2, stride=1)
         self.conv2 = nn.Conv2d(in_channels=6, out_channels=12, kernel_size=4, stride=2, padding=1)
-        self.pool2 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.pool2 = nn.MaxPool2d(kernel_size=2, stride=1)
         self.flatten = nn.Flatten()
-        self.fcn1 = nn.Linear(in_features=self.linear_line_size, out_features=120)
+
+        # Calculate linear size assuming fixed image size
+        self.linear_size = self.calculate_linear_size(image_size)
+
+        self.fcn1 = nn.Linear(in_features=self.linear_size, out_features=120)
+        self.dropout = nn.Dropout(p=0.2)
```

Comment:
Did you mean to add a hardcoded dropout here? Does it help performance noticeably?

Reply:
ah, no. This is an accidental commit, thanks for the catch. I'll remove it for now. If anything it should be added optionally based on a parameter